src/main/java/com/google/sps/app/servlets/AddUsersToEvent.java

```diff
@@ -0,0 +1,146 @@
+        List<String> usersToAdd = cleanInput(request.getParameter(USERS));
+        
+        //get lists
+        Entity eventEntity = getEventEntity(id);
+        List<String> usernames = getAssociatedEventUsernames(eventEntity);
+        List<String> userIDs = getAssociatedEventUserIDs(eventEntity);
+
+        //add to lists
+        UserList resp = addUsersToExistingUsers(usersToAdd, usernames, userIDs);
+
+        pushUserstoEvent(eventEntity, resp);
+
+        //return response
+        Gson gson = new Gson();
+        response.setContentType("appliation/json;");
+        response.getWriter().println(gson.toJson(resp));
+    }
+
+    private List<String> getAssociatedEventUserIDs(Entity eventEntity) {
+        List<Value<String>> datastoreList = eventEntity.getList("associatedUserIDs");
+        List<String> list = DataStoreHelper.convertToStringList(datastoreList);
+        return list;
+    }
+
+    private List<String> getAssociatedEventUsernames(Entity eventEntity) {
+        List<Value<String>> usernames = eventEntity.getList("associatedUsernames");
+        List<String> list = DataStoreHelper.convertToStringList(usernames);
+        return list;
+    }
+
+    private List<String> cleanInput(String input){
+        input = input.replace(" ", "");
+        String[] strArr = input.split(",");
+        return Arrays.asList(strArr);
+    }
+
+    private UserList addUsersToExistingUsers(List<String> usersToAdd, List<String> usernames, List<String> userIDs) {
+        UserList responseObj = new UserList(usernames, userIDs);
+
+        for (String userToAdd: usersToAdd){
+            String validationErrors = Validator.validateUserName(userToAdd);
+            if (!validationErrors.isEmpty()) {
+                responseObj.addError(String.format("Invalid %s for user %s", validationErrors, userToAdd));
+                continue;
+            } else if (usernames.contains(userToAdd)){
+                responseObj.addError(String.format("User %s already in list", userToAdd));
+                continue;
+            }
+
+            String userIDToAdd;
+            try {
+                userIDToAdd = DataStoreHelper.queryUserID(userToAdd);
+            } catch (IllegalArgumentException e) {
+                responseObj.addError(String.format("User %s not found", userToAdd));
+                continue;
+            }
+
+            //no error
+            responseObj.addUsername(userToAdd);
+            responseObj.addUserID(userIDToAdd);           
+
+        }
+
+        return responseObj;
+    }
+
+    private static Entity getEventEntity(String eventID){
+
+        Query<Entity> query =
+          Query.newEntityQueryBuilder()
+            .setKind("Event")
+            .setFilter(PropertyFilter.eq("eventID", eventID))
+            .build();
+        Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+        QueryResults<?> results = datastore.run(query);
+
+        return (Entity) results.next();
```

Comment:
Did you want to check here if the event exists?

Reply:
Done. I'm not sure if I should try to catch it because this error would indicate tampering with the session storage or a big mistake on our part.

Reply:
It is ok to throw an exception. You can throw a NotFoundException - https://docs.oracle.com/javaee/7/api/javax/ws/rs/NotFoundException.html